crates/mun_codegen/src/ir/dispatch_table.rs

```diff
@@ -150,16 +154,39 @@ struct TypedDispatchableFunction {
 
 impl<'a, D: IrDatabase> DispatchTableBuilder<'a, D> {
     /// Creates a new builder that can generate a dispatch function.
-    pub fn new(db: &'a D, module: &'a Module) -> Self {
-        DispatchTableBuilder {
+    pub fn new(
+        db: &'a D,
+        module: &'a Module,
+        intrinsics: BTreeMap<FunctionPrototype, FunctionType>,
+    ) -> Self {
+        let mut table = DispatchTableBuilder {
             db,
             module,
             function_to_idx: Default::default(),
             prototype_to_idx: Default::default(),
             entries: Default::default(),
             table_ref: None,
             table_type: module.get_context().opaque_struct_type("DispatchTable"),
+        };
+
+        if !intrinsics.is_empty() {
+            table.ensure_table_ref();
+
+            // Use a `BTreeMap` to guarantee deterministically ordered output
```

Comment:
Why is that important? A hashmap is also deterministic, right?

Reply:
A hasmap doesn't guarantee order of elements, so every run results in different IR due to differently inserted dispatch table entries. This also affects the snapshot tests and in general you probably want a deterministic compile result.

Reply:
:+1: